step.yml

```diff
@@ -83,6 +83,7 @@ inputs:
       - A Step in the workflow reads the commit history in order to generate a changelog
       - A Step in the workflow runs a git diff against a previous commit
 
+      Use the value `-1` to disable the depth limit completely and fetch the entire repo history.
```

Comment:
As I was reading your code I saw that in the `fetchConfig` struct we have two related properties:
```
struct fetchConfig {
    ...
    limitDepth bool
    depth         int
    ...
}
```
which made me think if instead having `-1` carry a special meaning we could have a separate field similar to the `limitDepth` variable in the example above. So it is just a bool field but maybe has a more precise meaning that adding `-1` value handling into the existing `clone_depth` field? 

I am not fully sure in it yet. I think I like the separate field better but just want to ask your opinion.

Reply:
I was contemplating the same, but I ended up with the -1 value because:

- There are already too many step inputs 😄 
- Disabling the depth limit is not something we recommend (for performance and network utilization reasons), so I'd rather not "advertise" it with the same weight as other step inputs. But if someone really needs the full repo history in CI, they can set this input to `-1`.

I can change my mind, but these are the reasons why I decided to represent the unlimited option in the same step input.